### PR TITLE
Potential topics for (not-only) FP meetup

### PR DESCRIPTION
**Pull request by mauriciojost**

### CODE REVIEW

**Comment by mauriciojost** [comment]
@mebubo ;) I will think about some other items I'd like to discuss.

**Comment by mauriciojost** [comment]
@generoso you too

**Comment by mauriciojost** [comment]
@mebubo @generoso I am not done for your information. I will update it with more items.

**Comment by mauriciojost** [comment]
I think I am done, I am leaving tomorrow so contact me on the group if any questions / clarifications. Feel free to add your own!

File: src/main/scala/org/mauritania/main4ino/Server.scala [hunk 1]
```diff
@@ -17,6 +17,10 @@ import org.mauritania.main4ino.security.AutherIO
 import scala.concurrent.ExecutionContext.Implicits.global
 import scala.concurrent.duration.FiniteDuration
 
+// If I replaced the below F=IO by some other Sync that does NOT delay the execution of side effects,
+// my program would stop having the property of referential transparency, correct? (side-effect expressions are evaluated
+// immediately without waiting for the end of the world).
+// What other Monad other than IO would it make sense to use here?
```
Comment (mebubo) [inline src/main/scala/org/mauritania/main4ino/Server.scala]: 1. Any worthy implementation of `Sync` should suspend execution of side effects.
2. This whole method could still be in `F`, and you could run the result of this method in `IO`. See https://github.com/pauljamescleary/scala-pet-store/blob/master/src/main/scala/io/github/pauljamescleary/petstore/Server.scala
3. Here you'd probably always use `IO`, but some of the instances for `IO` (e.g. `ApplicativeAsk`) could be provided by you manually.

**Pull request closed without merging**